src/WebJobs.Extensions.DurableTask/Options/AzureStorageOptions.cs

```diff
@@ -0,0 +1,122 @@
+        /// number of partitions.
+        /// </remarks>
+        /// <value>A positive integer between 1 and 16. The default value is <c>4</c>.</value>
+        public int PartitionCount { get; set; } = 4;
+
+        /// <summary>
+        /// Gets or sets the visibility timeout of dequeued control queue messages.
+        /// </summary>
+        /// <value>
+        /// A <c>TimeSpan</c> configured by the host. The default is 5 minutes.
+        /// </value>
+        public TimeSpan ControlQueueVisibilityTimeout { get; set; } = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Gets or sets the visibility timeout of dequeued work item queue messages.
+        /// </summary>
+        /// <value>
+        /// A <c>TimeSpan</c> configured by the host. The default is 5 minutes.
+        /// </value>
+        public TimeSpan WorkItemQueueVisibilityTimeout { get; set; } = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// Gets or sets the name of the Azure Storage connection string to use for the 
+        /// durable tracking store (History and Instances tables).
+        /// </summary>
+        /// <remarks><para>
+        /// If not specified, the <see cref="ConnectionStringName"/> connection string is used
+        /// for the durable tracking store.
+        /// </para><para>
+        /// This property is primarily useful when deploying multiple apps that need to share the same
+        /// tracking infrastructure. For example, when deploying two versions of an app side by side, using
+        /// the same tracking store allows both versions to save history into the same table, which allows
+        /// clients to query for instance status across all versions.
+        /// </para></remarks>
+        /// <value>The name of a connection string that exists in the app's application settings.</value>
+        public string TrackingStoreConnectionStringName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name prefix to use for history and instance tables in Azure Storage.
+        /// </summary>
+        /// <remarks>
+        /// This property is only used when <see cref="TrackingStoreConnectionStringName"/> is specified.
+        /// If no prefix is specified, the default prefix value is "DurableTask".
+        /// </remarks>
+        /// <value>The prefix to use when naming the generated Azure tables.</value>
+        public string TrackingStoreNamePrefix { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum queue polling interval.
+        /// </summary>
+        /// <value>Maximum interval for polling control and work-item queues.</value>
+        public TimeSpan MaxQueuePollingInterval { get; set; } = TimeSpan.FromSeconds(30);
+
+
+        internal override void ValidateHubName(string hubName)
+        {
+            try
+            {
+                NameValidator.ValidateBlobName(hubName);
```

Comment:
In your other PR, we're making it so that the task hub name has no default. In that case, should we be checking for a null-or-empty `hubName` here?

Reply:
The implementations of `CommonStorageProviderOptions.ValidateHubName()` are just going to be the storage provider specific limitations on task name. I restored the null check in the `DurableTaskOptions.Validate()` method, as that is a universal requirement.